src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  selectTodo: Todo;
+  setSelectTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ selectTodo, setSelectTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (selectTodo) {
+      setIsLoading(true);
+      getUser(selectTodo.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
+    }
```

Comment:
```suggestion
    setIsLoading(true);
    
    getUser(selectTodo.userId)
      .then(setUser)
      .finally(() => setIsLoading(false));
```
`selectTodo` is required prop